src/airstack/onchain_graph.py

```diff
@@ -0,0 +1,759 @@
+
+            if res.error is None:
+                followings = [following['followingAddress'] for following in (res.data.get(
+                    'SocialFollowings', {}).get('Following', []) or []) if 'followingAddress' in following]
+                recommended_users = format_farcaster_followings_data(
+                    followings,
+                    recommended_users
+                )
+
+                if not res.has_next_page:
+                    break
+                else:
+                    res = await res.get_next_page
+            else:
+                print("Error: ", res.error)
+                break
+
+        return recommended_users
+
+    async def __fetch_lens_followings(self, address, existing_users=[]):
+        social_followings_query = """
+        query MyQuery($user: Identity!) {
+        SocialFollowings(
+            input: {filter: {identity: {_eq: $user}, dappName: {_eq: lens}}, blockchain: ALL, limit: 200}
+        ) {
+            Following {
+            followingAddress {
+                addresses
+                domains {
+                name
+                isPrimary
+                }
+                socials {
+                dappName
+                blockchain
+                profileName
+                profileImage
+                profileTokenId
+                profileTokenAddress
+                }
+                xmtp {
+                isXMTPEnabled
+                }
+                mutualFollower: socialFollowers(
+                input: {filter: {identity: {_eq: $user}, dappName: {_eq: lens}}}
+                ) {
+                Follower {
+                    followerAddress {
+                    socials {
+                        profileName
+                    }
+                    }
+                }
+                }
+            }
+            }
+        }
+        }
+        """
+        res = None
+        recommended_users = existing_users.copy()
+        while True:
+            if res is None:
+                execute_query_client = self.api_client.create_execute_query_object(
+                    query=social_followings_query, variables={'user': address})
+                res = await execute_query_client.execute_paginated_query()
+
+            if res.error is None:
+                followings = [following['followingAddress'] for following in (res.data.get(
+                    'SocialFollowings', {}).get('Following', []) or []) if 'followingAddress' in following]
+                recommended_users = format_lens_followings_data(
+                    followings,
+                    recommended_users
+                )
+
+                if not res.has_next_page:
+                    break
+                else:
+                    res = await res.get_next_page
+            else:
+                print("Error: ", res.error)
+                break
+
+        return recommended_users
+
+    async def __fetch_farcaster_followers(self, address, existing_users=[]):
```

Comment:
You've used double underscores to indicate that some methods are private (e.g., __fetch_poaps_data). While this is a convention in Python, it's important to note that it doesn't make the methods truly private; it just changes their name to include the class name to avoid accidental name clashes in subclasses. If privacy is a concern, you should use a single underscore (e.g., _fetch_poaps_data) to indicate that the method is for internal use and shouldn't be accessed directly from outside the class.